web/js/codeworld.js

```diff
@@ -169,7 +169,12 @@ function initCodeworld() {
         }
     };
 
-    window.codeworldEditor.on('changes', window.updateUI);
+    window.reparseTimeoutId = null;
+    window.codeworldEditor.on("changes", () => {
+        if (window.reparseTimeoutId) clearTimeout(window.reparseTimeoutId);
+        window.reparseTimeoutId = setTimeout(parseSymbolsFromCurrentCode, 3000);
+        window.updateUI;
```

Comment:
This needs parentheses to call the function.

Reply:
done